### PR TITLE
Streaming renderer

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR that adds a streaming renderer. Main difference to `preact-ssr-prepass` is that it only awaits suspense boundaries and keeps everything else synchronous. This avoids a flooding the event loop with microtasks.

Tried a custom element approach to attach lazy hydrated subtrees, but the more I think about it the more I'm leaning towards using comments as folks sent me more and more use cases where introducing an arbitrary element breaks CSS styling (even with `display: contents`) or where the semantic HTML structure falls apart (`ul > li`).

Marking as draft until it's ready and we might need to make modifications to core too.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -295,20 +372,102 @@ function _renderToString(vnode, context, isSvgMode, selectValue, parent) {
...
+			vnode[PARENT] = undefined;
+
+			if (options.unmount) options.unmount(vnode);
+
+			return str;
+		} catch (error) {
+			if (renderer !== undefined && error.then) {
+				/** @type {import('./internal').Component} */
+				let component;
+				let susVNode = vnode;
```
Comment (jacob-ebey) [inline src/index.js:396]: Much sus, very amogus 😂

**Comment by Lucasvo1** [comment]
Any updates on this? 

It's a blocker for using Preact with Nextjs.

**Comment by osdiab** [comment]
Also a blocker for Remix, too.

**Comment by graysonhicks** [comment]
👋 This is also a blocker for `gatsby-plugin-preact` users trying to upgrade to Gatsby v5, which requires React 18.

**Comment by juliannehalversen** [comment]
Any updates here? As stated above this is a big issue for users trying to upgrade to Gatsby v5.

**Comment by developit** [comment]
Landing this PR doesn't make Preact support React's streaming renderer API, that would need to be plumbed through to [preact/compat/server](https://github.com/preactjs/preact/blob/master/compat/server.mjs).

**Comment by seo-burkedecor** [comment]
Hi, thank you so much for this fantastic project!

Do you guys have any update about what @developit explained?

**Comment by FleetAdmiralJakob** [comment]
Any updates here?

**Comment by xand3r40r93** [comment]
Any updates here?

**Comment by JoviDeCroock** [comment]
@AnderUstarroz you could help us test it https://github.com/preactjs/preact-render-to-string/pull/296 is the updated Pr and there's a linked Preact PR

**Comment by AnderUstarroz** [comment]
> @AnderUstarroz you could help us test it #296 is the updated Pr and there's a linked Preact PR

No problem @JoviDeCroock , just tell me which  versions of  `preact` and `preact-render-to-string` should I use to replace my current **package.json** dependencies:

```
  "dependencies": {
      "gatsby": "^5.12.4",
       "react": "^18.2.0",
       "react-dom": "^18.2.0",

      ...
```

**Comment by rschristian** [comment]
> > @AnderUstarroz you could help us test it #296 is the updated Pr and there's a linked Preact PR
> 
> No problem @JoviDeCroock , just tell me which versions of `preact` and `preact-render-to-string` should I use to replace my current **package.json** dependencies:

As Jovi mentioned, there's two open PRs. You'd have to build preact & rts from source.

**Comment by PavelAstapov** [comment]
Any updates? It is still a problem for Gatsby 5+ users

**Comment by Inviz** [comment]
Bummer that this isn't generally available yet. Looking forward

**Comment by JoviDeCroock** [comment]
Superseded by https://github.com/preactjs/preact-render-to-string/pull/296 and released in 6.5.0

**Pull request closed without merging**